### ISSUE TITLE
mBART fails to save checkpoint

### ISSUE THREAD

**Issue by odtuyzt** [pytorch/fairseq#3101]
## 🐛 Bug
Hi,

I am running an mBART model to summarize Turkish news on Google Colab. I have mostly followed the instructions at [the official example](https://github.com/pytorch/fairseq/tree/master/examples/mbart) but I have changed source language to news corpus to be summarized and target language to the summarized version of the news. When I begin to train my model it abruptly quits after this line:
2021-01-05 16:49:16 | INFO | fairseq.checkpoint_utils | Preparing to save checkpoint for epoch 1 @ 193 updates

then it prints:

^C.

This is keyboard interrupt however I do not invoke it.

I tried not saving a checkpoint and going on with epoch 2 and there was no problem. I also know the loss is decreasing and the sentencepiece tokenization seems fine when I inspect it.

### To Reproduce

Run this [notebook](https://colab.research.google.com/drive/1qEfBDFbzqNp_cD_WkgWNFlHT7GyzoxVt?usp=sharing). It will take around 15 minutes to load the necessary files and get to the error. 

#### Code sample

This is the training code:

!fairseq-train /content/pre/ \
  --encoder-normalize-before --decoder-normalize-before \
  --arch mbart_large --layernorm-embedding \
  --task translation_from_pretrained_bart \
  --source-lang src --target-lang tgt \
  --criterion label_smoothed_cross_entropy --label-smoothing 0.2 \
  --optimizer adam --adam-eps 1e-06 --adam-betas '(0.9, 0.98)' \
  --lr-scheduler polynomial_decay --lr 3e-05 --warmup-updates 25 --total-num-update 400 \
  --dropout 0.3 --attention-dropout 0.1 --weight-decay 0.0 \
  --max-tokens 256 --update-freq 2 \
  --save-interval 1 --save-interval-updates 5000 --keep-interval-updates -1 \
  --seed 222 --log-format tqdm --log-interval 8 \
  --restore-file /content/mbart.cc25.v2/model.pt \
  --reset-optimizer --reset-meters --reset-dataloader --reset-lr-scheduler \
  --langs ar_AR,cs_CZ,de_DE,en_XX,es_XX,et_EE,fi_FI,fr_XX,gu_IN,hi_IN,it_IT,ja_XX,kk_KZ,ko_KR,lt_LT,lv_LV,my_MM,ne_NP,nl_XX,ro_RO,ru_RU,si_LK,tr_TR,vi_VN,zh_CN \
  --ddp-backend no_c10d --memory-efficient-fp16  --checkpoint-suffix den

### Expected behavior

It should have saved a checkpoint and carried on with the training.

### Environment

 - fairseq Version : current one
 - PyTorch Version : 1.7.0+cu101
 - OS (e.g., Linux): Linux - Google Colab
 - How you installed fairseq (`pip`, source): pip from GitHub clone
 - Build command you used (if compiling from source):  
 - Python version: 3.6.9
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration: Tesla T4

**Comment by trina731** [comment]
This might be because Colab does not have enough memory to save the checkpoint. I was getting a similar error when Colab was low on RAM/disk space.
